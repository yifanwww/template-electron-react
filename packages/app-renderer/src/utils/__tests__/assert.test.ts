import { describe, expect, it } from '@jest/globals';

import { AssertionError, assert, assertIsNever } from '../assert';

describe(`Test fn \`${assert.name}\``, () => {
    it('should assert an expression', () => {
        const str = 'hello world';

        expect(() => assert(typeof str === 'string')).not.toThrow();
        expect(() => assert(typeof str === 'number')).toThrow(AssertionError);
        expect(() => assert(typeof str === 'number')).toThrowErrorMatchingSnapshot();
    });
});

describe(`Test fn \`${assertIsNever.name}\``, () => {
    it('should throw an error', () => {
        expect(() => assertIsNever('hello world' as never)).toThrow(AssertionError);
        expect(() => assertIsNever('hello world' as never)).toThrowErrorMatchingSnapshot();
    });
});
