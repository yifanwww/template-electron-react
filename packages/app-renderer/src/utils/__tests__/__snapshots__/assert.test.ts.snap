// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Test fn \`assert\` should assert an expression 1`] = `"Assertion Error."`;

exports[`Test fn \`assertIsNever\` should throw an error 1`] = `"hello world should be "never" type."`;
